File: DigitalwareFront/src/app/Models/factura.ts
import { Cliente } from './cliente';
import { FacturaProducto } from './factura-producto';

export class Factura {
  id: string;
  clienteId: string;
  total: number;
  fecha: string;
  FacturaProductos: FacturaProducto[];
  cliente: Cliente;
}

File: PruebaDigitalware.Models/Factura.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace PruebaDigitalware.Models
{
    public class Factura
    {
        public Guid Id { get; set; }
        public string ClienteId { get; set; }
        public long Total { get; set; }
        public DateTime Fecha { get; set; }

        [ForeignKey("FacturaId")]
        public List<FacturaProducto> FacturaProductos { get; set; }
        [NotMapped]
        public Cliente Cliente { get; set; }
    }
}
